static/app/utils/useExperiment.tsx

```diff
@@ -0,0 +1,84 @@
+import {useCallback, useEffect} from 'react';
+
+import {experimentConfig, unassignedValue} from 'sentry/data/experimentConfig';
+import ConfigStore from 'sentry/stores/configStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {
+  ExperimentAssignment,
+  ExperimentKey,
+  ExperimentType,
+  OrgExperiments,
+  UserExperiments,
+} from 'sentry/types/experiments';
+import {logExperiment as logExperimentAnalytics} from 'sentry/utils/analytics';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type UseExperimentOptions = {
+  /**
+   * By default this hook will log the exposure of the experiment upon mounting
+   * of the component.
+   *
+   * If this is undesirable, for example if the experiment is hidden behind
+   * some user action beyond this component being mounted, then you will want
+   * to customize when exposure to the experiment has been logged.
+   *
+   * NOTE: If set to false, YOU ARE RESPONSIBLE for logging exposure of the
+   *       experiment!! If you do not log exposure your experiment will not be
+   *       correct!!
+   */
+  logExperimentOnMount?: boolean;
+};
+
+type UseExperimentReturnValue<E extends ExperimentKey> = {
+  experimentAssignment: ExperimentAssignment[E];
+  /**
+   * Call this method when the user has been exposed to the experiment.
+   * You do not need to call this unless you have disabled logging on mount.
+   */
+  logExperiment: () => void;
+};
+
+function useExperimentAssignment(experiment: ExperimentKey) {
+  const organization = useOrganization();
+  const {user} = useLegacyStore(ConfigStore);
+  const {type} = experimentConfig[experiment];
+
+  if (type === ExperimentType.Organization) {
+    const key = experiment as keyof OrgExperiments;
+    return organization?.experiments?.[key] ?? unassignedValue;
```

Comment:
@malwilley Hmmm It seems like we should throw some errors if we have an experiment type as organization/user but the organization/user is falsy

Reply:
I realize `useOrganization` is going to blow up if org is falsy as well.

Reply:
Good thinking! Added some Sentry captureMessage's for when the data doesn't match up correctly.

Do you think there are use cases where we want to use this outside the org context?

Reply:
@malwilley I can't think of any right now. I think it's fine to assume the org context is there.